### PR TITLE
[JAX] Fully remove attn_type and set self_attn_mask_type default to 'causal'

### PR DESCRIPTION
**Pull request by zlsh80826**
Background:
[PR-273:Add self_attn_mask_type and replace attn_type](https://github.com/NVIDIA/TransformerEngine/pull/273) introduced `self_attn_mask_type` and planed to fully remove `attn_type` in v0.11.

This PR does the following thing as PR-273 follow up:
1. Fully remove `attn_type`
2. Set the default value of `self_attn_mask_type` to `"causal"` to align pytorch framework

### CODE REVIEW

File: transformer_engine/jax/flax/transformer.py [hunk 1]
```diff
@@ -877,6 +858,7 @@ class TransformerLayer(nn.Module):
         Transformer in conjunction with the TransformerLayerType.ENCODER option.
     self_attn_mask_type: {'causal', 'padding'}, default = 'causal'
         Type of attention mask passed into softmax operation.
+        New in version 0.10.
```
Comment (ksivaman) [inline transformer_engine/jax/flax/transformer.py:861]:
```suggestion
        Introduced in v0.10.0.
```

File: transformer_engine/jax/flax/transformer.py [hunk 2]
```diff
@@ -244,11 +244,9 @@ class MultiHeadAttention(nn.Module):
         Indicate if apply residual connection with the output of layer normalization.
     output_layernorm : bool, default = False
         Indicate if apply a layer normalization at the end of MHA.
-    attn_type: Any, defult = None
-        *Deprecated*, will be ignored in v0.10 and be fully removed in v0.11.
-        Please use `attn_mask_type` to config the attention mask.
     attn_mask_type: {'causal', 'padding'}, default = 'causal'
         Type of attention mask passed into softmax operation.
+        New in version 0.10.
```
Comment (ksivaman) [inline transformer_engine/jax/flax/transformer.py:249]:
```suggestion
        Introduced in v0.10.0.
```

**Review by ksivaman — APPROVED** [review]
LGTM aside from minor doc suggestions

**Review by timmoon10 — APPROVED** [review]

**Pull request merged**